packages/tools/bundleDefaultPlugins.ts

```diff
@@ -0,0 +1,87 @@
+import path = require('path');
+import { execCommand2 } from './tool-utils';
+import fs = require('fs-extra');
+const fetch = require('node-fetch');
+const util = require('util');
+const streamPipeline = util.promisify(require('stream').pipeline);
+
+interface PluginAndVersion {
+   [pluginId: string]: string;
+}
+
+export const localPluginsVersion = async (defaultPluginDir: string, defaultPluginsId: PluginAndVersion): Promise<PluginAndVersion> => {
+	if (!(await fs.pathExists(path.join(defaultPluginDir)))) await fs.mkdirp(defaultPluginDir);
+
+	const localPluginsVersions: PluginAndVersion = {};
+
+	for (const pluginId of Object.keys(defaultPluginsId)) {
+
+		if (!(await fs.pathExists(`${defaultPluginDir}/${pluginId}`))) {
+			localPluginsVersions[pluginId] = '0.0.0';
+			continue;
+		}
+		const data = fs.readFileSync(`${defaultPluginDir}/${pluginId}/manifest.json`, 'utf8');
+		const manifest = JSON.parse(data);
+		localPluginsVersions[pluginId] = manifest.version;
+	}
+	return localPluginsVersions;
+};
+
+const downloadFile = async (url: string, name: string) => {
+	const response = await fetch(url);
+	if (!response.ok) throw new Error(`Unexpected response ${response.statusText}`);
+	await streamPipeline(response.body, fs.createWriteStream(name));
```

Comment:
Is it not possible to use fetch? Too many problems with streams in general

Reply:
I think we can't use `fetch` alone, we have to write the response to a file. If we don't want to use streams then we can use the following alternative:

```
function downloadFile(url:string, outputPath:string) {
	return fetch(url)
		.then(response => response.buffer())
		.then(buff => writeFilePromise(outputPath, Buffer.from(buff)));
}
```